nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,56 @@
+const { response, request } = require("express");
+const express = require("express");
+const router = express.Router();
+
+const meals = require("../data/meals.json");
+
+// 12 tests passed, 6 failed. Can't get my code to work 
+
+router.get("/", async (request, response) => {
+  try {
+    console.log(request.query.maxPrice);
+    response.send(meals);
+  } catch (error) {
+    throw error;
+  }
+});
+
+
+// copied from app.js(week1)
+router.get("/meet", async (request, response) => {
+  response.send("Meal Sharing Web APP");
+});
+
+
+router.get("/cheap-meals", (request, response) => {
+  cheapMeals = meals.filter((cheapmeal) => {
+    return cheapmeal.price < 50;
+  });
+  response.json(cheapMeals);
+});
+
+router.get("/large-meals", async (request, response) => {
+  largeMeals = meals.filter((largeMeal) => {
+    return largeMeal.maxNumberOfGuests > 3;
+  });
+  response.json(largeMeals);
+});
+
+router.get("/meal", async (request, response) => {
+  response.send("Meal Sharing Web App");
+});
```

Comment:
It seems like you are missing big part of this week's homework here. Look at the description of the homework, and you'll see what routes and query parameters you should implement.

Reply:
well, I'm still trying to get all tests succeed, I get 12 passed and 6 failed. and yet I can't figure it out.